core/services/route_match/RouteMatchSpecificationManager.php

```diff
@@ -0,0 +1,146 @@
+     */
+    public function initialize()
+    {
+        $this->populateSpecificationCollection();
+    }
+
+
+    /**
+     * @return CollectionInterface|RouteMatchSpecificationInterface[]
+     * @throws CollectionLoaderException
+     * @throws CollectionDetailsException
+     * @since $VID:$
+     */
+    private function populateSpecificationCollection()
+    {
+        $loader = new CollectionLoader(
+            new CollectionDetails(
+            // collection name
+                RouteMatchSpecificationCollection::COLLECTION_NAME,
+                // collection interface
+                'EventEspresso\core\domain\entities\route_match\RouteMatchSpecificationInterface',
+                // FQCNs for classes to add (all classes within each namespace will be loaded)
+                apply_filters(
+                    'FHEE__EventEspresso_core_services_route_match_RouteMatchSpecificationManager__populateSpecificationCollection__collection_FQCNs',
+                    array(
+                        'EventEspresso\core\domain\entities\route_match\specifications\admin',
+                        // 'EventEspresso\core\domain\entities\route_match\specifications\public',
+                    )
+                ),
+                // filepaths to classes to add
+                array(),
+                // file mask to use if parsing folder for files to add
+                '',
+                // what to use as identifier for collection entities
+                // using CLASS NAME prevents duplicates (works like a singleton)
+                CollectionDetails::ID_CLASS_NAME
+            ),
+            $this->specifications,
+            null,
+            $this->specifications_factory
+        );
+        return $loader->getCollection();
+    }
+
+
+    /**
+     * Given the FQCN for a RouteMatchSpecification, will return true if the current request matches
+     *
+     * @param string $routeMatchSpecificationFqcn fully qualified class name
+     * @return bool
+     * @throws InvalidClassException
+     * @since $VID:$
+     */
+    public function currentRequestMatches($routeMatchSpecificationFqcn)
+    {
+        /** @var RouteMatchSpecificationInterface $specification */
+        $specification = $this->specifications->get($routeMatchSpecificationFqcn);
+        if ( ! $specification instanceof $routeMatchSpecificationFqcn) {
+            throw new InvalidClassException($routeMatchSpecificationFqcn);
+        }
+        return $specification->routeMatches();
+    }
+
+
+    /**
+     * Handy method for development that returns
+     * a list of existing RouteMatchSpecification classes
+     * matching the current request that can be used to identify it.
+     * If no matches are returned (or nothing acceptable)
+     * then create a new one that better matches your requirements.
+     *
+     * @return array
+     * @since $VID:$
+     */
+    public function findSpecificationsMatchingCurrentRequest()
```

Comment:
I'd prefer either `findRouteSpecificationsMatchingCurrentRequest` or `findRoutesMatchingCurrentRequest` as a method name.

Reply:
will change to `findRouteSpecificationsMatchingCurrentRequest`